analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp9.cs

```diff
@@ -241,3 +242,52 @@ static void Demo(object demo)
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/8011
+namespace Repro_8011
+{
+    [Flags]
+    public enum Flags
+    {
+        None,
+        Foo,
+        Bar
+    }
+
+    public class VarWhen
+    {
+        public int Test(Flags flags)
+        {
+            return flags switch
+            {
+                var value when value.HasFlag(Flags.Foo) => 1, // Noncompliant FP
```

Comment:
Please assert also the message for these issues.
It will speed up our understanding upon tackling it.